### PR TITLE
*: update for neo3 preview2

### LINKED ISSUE #17: Update to neo3-preview2
Branch out master-2.x version of the workshop and update it to current master of neo-go. Multisig operations are gonna be needed at the start of the workshop, other than that it should be a simple compile & run.

### PR DESCRIPTION
**Pull request by AnnaShaleva**
Closes #17

* Update workshop documentation
* Update smart-contracts and configurations
* Update wallet

### CODE REVIEW

File: 4-domain.go [hunk 1]
```diff
@@ -7,104 +7,125 @@ import (
...
+func Register(domainName []byte, owner []byte) bool {
+	ctx := storage.GetContext()
+
+	if !runtime.CheckWitness(owner) {
+		runtime.Log("Owner argument is not the same as the sender")
+		return false
+	}
+
+	exists := storage.Get(ctx, domainName)
+	if exists != nil {
```
Comment (roman-khimov) [inline 4-domain.go]: `if storage.Get(ctx, domainName) != nil {}`
Make this code a bit more efficient as you don't really need a variable to store the result.

File: 4-domain.go [hunk 2]
```diff
@@ -7,104 +7,125 @@ import (
...
+func Transfer(domainName []byte, toAddress []byte) bool {
+	ctx := storage.GetContext()
+	owner := storage.Get(ctx, domainName)
+	if owner == nil {
+		runtime.Log("Domain is not yet registered")
+		return false
+	}
+
+	if !runtime.CheckWitness(owner.([]byte)) {
+		runtime.Log("Sender is not the owner, cannot transfer")
```
Comment (roman-khimov) [inline 4-domain.go]: `Not co-signed by an owner...` would be more appropriate for Neo 3.

File: README.md [hunk 3]
```diff
@@ -116,11 +116,11 @@ INDEX    OPCODE          PARAMETER
 37       RET     
 ```
 Later we will use this compiled contract in a workshop =).
-More information how to use cli you can find [here](https://github.com/nspcc-dev/neo-go/blob/master-2.x/docs/cli.md)
+More information how to use cli you can find [here](https://github.com/nspcc-dev/neo-go/blob/master/docs/cli.md)
```
Comment (roman-khimov) [inline README.md]: `You can find more information on how to use the CLI here` or `more info ... can be found here`.

File: README.md [hunk 4]
```diff
@@ -236,75 +237,82 @@ $ ./bin/neo-go node --privnet
...
+2020-06-30T16:26:47.550+0300	INFO	starting rpc-server	{"endpoint": ":20331"}
+2020-06-30T16:26:47.550+0300	INFO	node started	{"blockHeight": 0, "headerHeight": 0}
+2020-06-30T16:26:47.551+0300	INFO	new peer connected	{"addr": "127.0.0.1:20333", "peerCount": 1}
+2020-06-30T16:26:47.551+0300	INFO	new peer connected	{"addr": "127.0.0.1:20334", "peerCount": 2}
+2020-06-30T16:26:47.552+0300	INFO	new peer connected	{"addr": "127.0.0.1:20335", "peerCount": 3}
+2020-06-30T16:26:47.552+0300	INFO	new peer connected	{"addr": "127.0.0.1:20336", "peerCount": 4}
+2020-06-30T16:26:47.553+0300	INFO	started protocol	{"addr": "127.0.0.1:20334", "userAgent": "/NEO-GO:/", "startHeight": 3, "id": 2280194870}
+2020-06-30T16:26:47.554+0300	INFO	started protocol	{"addr": "127.0.0.1:20333", "userAgent": "/NEO-GO:/", "startHeight": 3, "id": 3666776256}
+2020-06-30T16:26:47.555+0300	INFO	started protocol	{"addr": "127.0.0.1:20336", "userAgent": "/NEO-GO:/", "startHeight": 3, "id": 1699156200}
+2020-06-30T16:26:47.555+0300	INFO	started protocol	{"addr": "127.0.0.1:20335", "userAgent": "/NEO-GO:/", "startHeight": 3, "id": 874998449}
```
Comment (roman-khimov) [inline README.md]: `NEO-GO:`?

File: README.md [hunk 5]
```diff
@@ -479,84 +473,87 @@ Which means that our contract was deployed and now we can invoke it.
...
-      "txid" : "0x6f27f523da9c71297f4a81a254274b0c2a78f893b81c500429be6230254be0bf"
-   },
-   "jsonrpc" : "2.0"
+      ]
+   }
 }
 ```
 Pay attention to `notification` field. It contains messages, which where passed to `runtime.Notify` method.
-This one contains hexadecimal byte arrays which can be decoded into 3 messages:
+This one contains base64 byte arrays which can be decoded into 3 messages:
```
Comment (roman-khimov) [inline README.md]: Maybe we should show some `echo string | base64 -d` here.

File: README.md [hunk 6]
```diff
@@ -652,344 +652,319 @@ Let’s go!
...
 In other words, we should transfer all available amount of token (total supply) to someone's account.
-There's a special function for this purpose in our contract - `mint` function, so let's mint token to our address:
+There's a special function for this purpose in our contract - `Mint` function, so let's mint token to our address:
 ```
-./bin/neo-go contract invokefunction -e http://localhost:20331 -w my_wallet.json -g 0.00001 f84d6a337fbc3d3a201d41da99e86b479e7a2554 mint AK2nJJpJr6o664CWJKi1QRXjqeic2zRp8y
+$ ./bin/neo-go contract invokefunction -r http://localhost:20331 -w my_wallet.json 081480331bcd3183cd6c4d0c035717bba1b4daae Mint NMipL5VsNoLUBUJKPKLhxaEbPQVCZnyJyB -- f0a33acbf435417f3d2c18445d607d1e6f48d413
 ```
+Where:
+- `--` is a special delimiter of transaction's cosigners list
+- `f0a33acbf435417f3d2c18445d607d1e6f48d413` is the cosigner itself (which is hex-encoded LE representation of our `NMipL5VsNoLUBUJKPKLhxaEbPQVCZnyJyB` account)
```
Comment (roman-khimov) [inline README.md:910]: Maybe we should add a word or two about cosigners here.

File: README.md [hunk 7]
```diff
@@ -652,344 +652,319 @@ Let’s go!
...
+      "notifications" : []
    },
    "id" : 1,
    "jsonrpc" : "2.0"
 }
 ```
-Now we can see non-empty byte array at the `stack` field, so `00c040b571e803` is a hexadecimal representation of the nep5 token balance of our account.
+Now we can see integer value at the `stack` field, so `1100000000000000` is the nep5 token balance of our account.
 
 Note, that token can be minted only ones.
```
Comment (roman-khimov) [inline README.md]: Not related, but `once`.

File: README.md [hunk 8]
```diff
@@ -652,344 +652,319 @@ Let’s go!
...
+      "trigger" : "Application",
+      "vmstate" : "HALT",
+      "notifications" : []
    },
-   "id" : 1,
    "jsonrpc" : "2.0"
 }
 ```
-Here we are! There are exactly 5 tokens at the `stack` field. You can also ensure that these 5 tokens were debited from `AK2nJJpJr6o664CWJKi1QRXjqeic2zRp8y` account by using `balanceOf` method.
+Here we are! There are exactly 5 tokens at the `stack` field. You can also ensure that these 5 tokens were debited from `NMipL5VsNoLUBUJKPKLhxaEbPQVCZnyJyB` account by using `BalanceOf` method.
```
Comment (roman-khimov) [inline README.md]: It just occured to me that this actually breaks NEP5. It must be `balanceOf` to conform the standard.

Comment (roman-khimov) [reply README.md]: And the other methods have the same issue.

File: README.md [hunk 9]
```diff
@@ -1119,14 +1089,14 @@ import (
 )
 
 func main() {
-	bytes, _ := hex.DecodeString("72656769737465726564")
-	fmt.Println(string(bytes))
+	bytes, _ := base64.StdEncoding.DecodeString("cmVnaXN0ZXJlZA==")
+    fmt.Println(string(bytes))
```
Comment (roman-khimov) [inline README.md]: Not needed any longer, use standard `base64` command (it's a part of coreutils, so it's widely available).

**Review by roman-khimov — CHANGES_REQUESTED** [review]
There should be some link to 2.x version of the workshop, the README is shown on the main repository page.

**Comment by AnnaShaleva** [comment]
> There should be some link to 2.x version of the workshop, the README is shown on the main repository page.

Added `Versioning` part at the start.

File: README.md (hunk 5 above)
Comment (AnnaShaleva) [reply README.md]: Did you mean adding a couple of words on how to use `echo string | base64 -d` or just putting application log with human-readable fields?

Comment (roman-khimov) [reply README.md]: We should show how to decipher these base64 values, application log is application log, it shouldn't be changed in any way.

File: README.md (hunk 9 above)
Comment (AnnaShaleva) [reply README.md]: I can not find it, could you, please, give a full method name?

Comment (roman-khimov) [reply README.md]: No methods, pure CLI.
```
$ echo ... | base64 -d
```
Way easier than this helper in Go.

File: README.md (hunk 4 above)
Comment (AnnaShaleva) [reply README.md]: It's something in neo-go, I'll check this.

File: README.md [hunk 10]
```diff
@@ -1152,158 +1199,148 @@ All of these values let us be sure that our domain was successfully registered.
...
-- First one with `64656c65746564` byte array, which is `deleted` with additional information (domain `my_first_domain` was deleted from account `AK2nJJpJr6o664CWJKi1QRXjqeic2zRp8y`),
-- Second one with `72656769737465726564` byte array, which is `registered` with additional information  (domain `my_first_domain` was registered with account `AKkkumHbBipZ46UMZJoFynJMXzSRnBvKcs`).
+- First one with `ZGVsZXRlZA==` byte string, which is `deleted` with additional information (domain `my_first_domain` was deleted from account `NMipL5VsNoLUBUJKPKLhxaEbPQVCZnyJyB`),
+- Second one with `cmVnaXN0ZXJlZA==` byte string, which is `registered` with additional information  (domain `my_first_domain` was registered with account `NgzuJWWGVEwFGsRrgzj8knswEYRJrTe7sm`).
 The `stack` field contains `1` value, which means that our domain was successfully transferred.
 
 #### Step #6
 
-The last call is `delete`, so you can try to create the other domain, e.g. `my_second_domain` and then remove it from storage with:
+The last call is `Delete`, so you can try to create the other domain, e.g. `my_second_domain` and then remove it from storage with:
```
Comment (roman-khimov) [inline README.md]: Isn't it lowercase now?

Comment (AnnaShaleva) [reply README.md]: Yes, I missed this. Fixed

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**